### ISSUE TITLE
Default project-specific hub interface configuration

### ISSUE THREAD

**Issue by mjy** [SpeciesFileGroup/taxonworks#19]
This was envisioned by both MM and DM.

Task and data tabs within the hub are currently defaulted to the configuration defined in https://github.com/SpeciesFileGroup/taxonworks/tree/development/config/interface/hub.  Projects should be able to provide there own .yml configurations for each of data/tasks.  

Rather than provide a user-interface to all the myriad options, at present the easiest way to do this is to allow a project admin to load their own .yml files, and we persist these as jsonb hashes in the @project record.  Since the configuration is just a hash it will be trivial to modify the hub code to use the project has rather than the TW default.  The major code needed is a validator(s) for two yml files.  With them in place we just need a file-field form that has a setter that writes to the jsonb project attribute.

Steps to implement:
* Write module that validates user_tasks.yml in general
* Write module that validates data.yml in general
* Add migration to create jsonb attribute columns ```data_yml_settings``` and ```user_tasks_yml_settings``` (requires postgresql 4.?, this is OK, we need jsonb for core features)
* Create project admin form to accept user-defined data.yml and user_tasks.yml. Show user validation errors if files not valid.
* Write a application level method that returns the hub configuration, defaulting to user provided values if provided, falling back to existing methods if not

**Comment by mjy** [comment]
This issue needs to be refactored into a general 'about:config' type framework that has several different base namespaces.  Early idea from @jlpereira:

```
{
    "TW::Workbench::Pinboard::Expanded::TaxonNames": true,
    "TW::Workbench::Pinboard::Expanded::Otus": true,
    "TW::Workbench::Pinboard::Expanded::Documentation": false,
}
```

**Comment by mjy** [comment]
With the scope of the overall project not everything is going to fit on one
screen, and *some* configuration will always be necessary, as workflows and
priorities change throughout a project.  The favourites mechanism is one
way to faciliate that evolution, it can definitely evolve, but I don't see
an all in one solution as practical either.

**Comment by anadm** [comment]
agreed - maybe the search on the right could include both tasks and data 
definitely will need to pick default favorites for home screens
